framework/src/base/FEProblem.C

```diff
@@ -1820,20 +1828,28 @@ FEProblem::addMaterial(const std::string & mat_name, const std::string & name, I
       object_name = name + "_neighbor";
       MooseSharedPointer<Material> neighbor_material = _factory.create<Material>(mat_name, object_name, current_parameters, tid);
 
+      // dirac material
+      current_parameters.set<Moose::MaterialDataType>("_material_data_type") = Moose::DIRAC_MATERIAL_DATA;
+      current_parameters.set<bool>("_dirac") = true;
+      object_name = name + "_dirac";
+      MooseSharedPointer<Material> dirac_material = _factory.create<Material>(mat_name, object_name, current_parameters, tid);
```

Comment:
This is a no-go.  It's adding a burden to every MOOSE solve out there... for the tiny possibility of doing something that has never even been allowed for the last 8 years.

If you want to enable DiracKernels to have some access to stateful material properties... you're going to need to do it "on demand" somehow...